src/EVM/Expr.hs

```diff
@@ -663,6 +663,8 @@ simplify e = if (mapExpr go e == e)
     go (WriteWord a b c) = writeWord a b c
 
     go (WriteByte a b c) = writeByte a b c
+    go (CopySlice srcOff@(Lit 0) dstOff size@(Lit sz) (WriteWord (Lit 0) value (ConcreteBuf buf)) dst) = (CopySlice srcOff dstOff size (WriteWord (Lit 0) value (ConcreteBuf simplifiedBuf)) dst)
```

Comment:
how hard would it be to extend this to work for all literals src offsets?

Reply:
Yeah, it'd be possible :D I am always terrified of overflows. I'll try to figure the overflows it out. It can be quite painful :D I'll get back to this tomorrow.

Reply:
I have now updated this to be more generic. It is more generic in every way. I put a guard to make sure it does not overflow.